### PR TITLE
E2E testing

### PR DESCRIPTION
**Pull request by janosh**
Another step towards closing #263 (bullet point 3). This PR adds more end-to-end testing with `dash.testing`.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,297 +1,4 @@
 # macOS
 .DS_Store
 
-# all PyCharm settings
```
Comment (janosh) [inline .gitignore:4]: @mkhorton There seemed to be a lot of unneeded ignores in here. Let me know if I removed anything that was important.

Comment (mkhorton) [reply .gitignore:4]: I think this was from a standard .gitignore template from somewhere, but I'm fine removing

**Comment by mkhorton** [comment]
This looks great, thank you! Added a brief introductory section to the new doc page.

**Comment by janosh** [comment]
Wow, that was quick! This was actually still WIP:
`poetry` caching isn't working yet and I was going to try to resolve [these CI errors](https://github.com/materialsproject/crystaltoolkit/runs/7549568924?check_suite_focus=true#step:7:305)

```py
requests.exceptions.HTTPError: 409 Client Error: Conflict for url: https://percy.io/api/v1/builds/
```


Should have marked as draft, sorry. Anyway, I'll carry on in another PR.

**Comment by mkhorton** [comment]
No worries -- thanks, yes I definitely assumed this was complete! What I reviewed looked ok.

There was just an update to `poetry` caching I think in the last few days (there was an open issue regarding issues with caches not being updated as they should be), not sure if it'll affect what you were looking at.

Percy is great, but hasn't been used in a while. I'd be glad to get it working again but not sure if they've had API changes since we were last using it. Let me know if you need any API keys re-generated etc.

**Comment by janosh** [comment]
I think `poetry` caching is working now after https://github.com/materialsproject/crystaltoolkit/commit/e2b3c5a366f3bb98b77ac1271a08dafa5b17a582.

Just to make sure, I understood #263 correctly, the goal is to have all example apps run and tested in CI, right? I'll do that in the next PR then.

**Comment by mkhorton** [comment]
Yes; to clarify though, I wasn't actually expecting you to get the CI working, just to make sure that any relevant test(s) for the example apps were in place (i.e., that they ran fine locally, that they tested what was appropriate to test, that screenshots were taken etc). Nevertheless, if you're ok working on the CI, that's much appreciated.

**Pull request merged**